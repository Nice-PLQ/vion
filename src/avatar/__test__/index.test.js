import { mount } from '@vue/test-utils';
import Avatar from '../avatar.vue';

describe('Avatar', () => {
  const wrapper = mount(Avatar, {
    props: {
      url: 'https://path/to/avatar/image.jpg',
      size: '36px'
    }
  });

  test('单一头像', () => {
    expect(wrapper.classes()).not.toContain('v-avatar_2');
    expect(wrapper.element.style.fontSize).toBe('36px');
    expect(wrapper.find('img').attributes('src')).toBe('https://path/to/avatar/image.jpg');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('头像组合', async () => {
    await wrapper.setProps({
      url: [
        'https://path/to/avatar/image_1.jpg',
        'https://path/to/avatar/image_2.jpg',
        'https://path/to/avatar/image_3.jpg',
      ]
    });
    expect(wrapper.classes()).toContain('v-avatar_3');
    expect(wrapper.element).toMatchSnapshot();
  });

  test('click事件触发', async () => {
    await wrapper.trigger('click');
    expect(wrapper.emitted('click')).toBeTruthy();
  });

  test('轮播头像', async () => {
    const wrapper = mount(Avatar, {
      props: {
        url: [
          'https://path/to/avatar/image_1.jpg',
          'https://path/to/avatar/image_2.jpg',
          'https://path/to/avatar/image_3.jpg',
        ],
        size: '36px',
        carousel: true,
      },
    });
    expect(wrapper.classes()).toContain('v-avatar_carousel');
    expect(wrapper.element).toMatchSnapshot();
  });
});
