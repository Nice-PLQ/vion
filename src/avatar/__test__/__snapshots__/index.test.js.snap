// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar 单一头像 1`] = `
<div
  class="v-avatar"
  style="font-size: 36px;"
>
  
  
  <img
    src="https://path/to/avatar/image.jpg"
  />
  
  
  
  
</div>
`;

exports[`Avatar 头像组合 1`] = `
<div
  class="v-avatar v-avatar_3"
  style="font-size: 36px;"
>
  
  
  <img
    src="https://path/to/avatar/image_1.jpg"
  />
  <img
    src="https://path/to/avatar/image_2.jpg"
  />
  <img
    src="https://path/to/avatar/image_3.jpg"
  />
  
  
  
  
</div>
`;

exports[`Avatar 轮播头像 1`] = `
<div
  class="v-avatar v-avatar_carousel"
  style="font-size: 36px;"
>
  <transition-group-stub
    appear="false"
    css="true"
    name="horizon-fade-zoom"
    persisted="false"
  >
    
    <img
      src="https://path/to/avatar/image_1.jpg"
    />
    <img
      src="https://path/to/avatar/image_2.jpg"
      style="display: none;"
    />
    <img
      src="https://path/to/avatar/image_3.jpg"
      style="display: none;"
    />
    
  </transition-group-stub>
  
  
</div>
`;
